Repository/AbstractPhpcrRepository.php

```diff
@@ -72,6 +82,58 @@ public function find($query, $language = 'glob')
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function remove($query, $language = 'glob')
+    {
+        $this->failUnlessGlob($language);
+        $nodes = $this->finder->find($this->resolvePath($query));
+
+        // if the query does not contain a glob pattern, then it applies to an
+        // explicit node and we should throw an exception if it is not found.
+        if (false === $this->globHelper->isGlobbed($query) && 0 === count($nodes)) {
+            throw new ResourceNotFoundException($query);
```

Comment:
Not sure if we should throw an exception if an explicit node was specified but not found (as here) or always throw an exception if zero nodes are found. /cc @WouterJ @webmozart